core/src/main/java/io/github/belgif/rest/guide/validator/core/model/PathDefinition.java

```diff
@@ -14,11 +14,15 @@ public class PathDefinition extends OpenApiDefinition<PathItem> {
 
     public PathDefinition(PathItem model, PathsDefinition parent, String identifier) {
         super(model, parent, identifier, JsonPointer.relative(identifier));
-        this.isDirectPath = parent.isInMainFile();
+        this.isDirectPath = parent.isInMainFile() && !pathsUsedAsRefsOnly();
     }
 
     @Override
     public PathItem getModel() {
         return super.getModel();
     }
+
+    private boolean pathsUsedAsRefsOnly() {
```

Comment:
Java convention to start methods with get/is/hasXXX
+ align with renamed x-reusable-definitions-only
=> hasReusableDefinitionsOnly()